lib/orderbook/OrderBook.ts

```diff
@@ -98,7 +98,14 @@ class OrderBook extends EventEmitter {
           this.emit('ownOrder.swapped', { pairId, quantity, id: orderId });
         }
       });
-      // TODO: bind to other swap events
+      this.swaps.on('swap.failed', (deal) => {
+        if (deal.role === SwapRole.Maker && deal.phase === SwapPhase.SwapAgreed) {
+          // if our order is the maker and the swap failed after it was agreed to but before it was executed
+          // we must release the hold on the order that we set when we agreed to the deal
+          const order = this.getOwnOrder(deal.orderId, deal.pairId);
+          order.hold! -= deal.quantity!;
```

Comment:
this breaks the repository encapsulation.

Reply:
I don't get what you mean, could you elaborate? Do you mean this line should be in TradingPair.ts?

Reply:
Yes, I means repository in the greater sense, not our db repository class. `TradingPair` now manages the orders in-memory, so all the CRUD operations should be done through its API.

Reply:
Alright that makes sense, thanks for clarifying.